yellow-paper/docs/public-vm/avm.md

```diff
@@ -302,9 +307,42 @@ nestedMachineState = MachineState {
+
 > Note: recall that `INITIAL_MESSAGE_CALL_RESULTS` is the same initial value used during [context initialization for a public execution request's initial message call](#context-initialization-for-initial-call).
 > `STATICCALL_OP` and `DELEGATECALL_OP` refer to the 8-bit opcode values for the `STATICCALL` and `DELEGATECALL` instructions respectively.
 
-### Updating the calling context after nested call halts
\ No newline at end of file
+### Updating the calling context after nested call halts
+When a message call's execution encounters an instruction that itself triggers a message call, the nested call executes until it reaches a halt. At that point, control returns to the caller, and the calling context is updated based on the sub-context and the message call instruction's transition function. The components of that transition function are defined below.
+
+The success or failure of the nested call is captured into memory at the offset specified by the call instruction's `successOffset` input:
+```
+context.machineState.memory[instr.args.successOffset] = subContext.results.reverted
+```
+
+Recall that a nested call is allocated some gas. In particular, the call instruction's `gasOffset` input points to an L1 and L2 gas allocation for the nested call. As shown in the [section above](#context-initialization-for-a-nested-call), a nested call's `subContext.machineState.l1GasLeft` is initialized to `context.machineState.memory[instr.args.gasOffset]`. Likewise, `l2GasLeft` is initialized from `gasOfffset+1`.
+
+As detailed in [the gas section above](#gas-cost-notes-and-examples), every instruction has an associated `instr.l1GasCost` and `instr.l2GasCost`. A nested call instruction's cost is the same as its initial `l*GasLeft` and `l2GasLeft`. Prior to the nested call's execution, this cost is subtracted from the calling context's remaining gas.
+
+When a nested call completes, any of its allocated gas that remains unused is refunded to the caller.
+```
+context.l1GasLeft += subContext.machineState.l1GasLeft
+context.l2GasLeft += subContext.machineState.l2GasLeft
+```
+
+If a nested call halts normally with a [`RETURN`](./InstructionSet/#isa-section-return) or [`REVERT`](./InstructionSet/#isa-section-revert), it may have some output data (`subContext.results.output`). The caller's `retOffset` and `retSize` arguments to the nested call instruction specify a region in memory to place output data when the nested call returns.
+```
+if instr.args.retSize > 0:
+    context.memory[instr.args.retOffset:instr.args.retOffset+instr.args.retSize] = subContext.results.output
+```
+
+As long as a nested call has not reverted, its updates to the world state and accrued substate will be absorbed into the calling context.
+```
+if !subContext.results.reverted and instr.opcode != STATICCALL_OP:
```

Comment:
nitpick: I like to capitalize the logical AND. I think it helps with readability.